prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/registry/MultiCollector.java

```diff
@@ -34,6 +44,23 @@ default MetricSnapshots collect(Predicate<String> includedNames) {
         return result.build();
     }
 
+    /**
+     * Like {@link #collect(includedNames)}, but with support for multi-target pattern.
+     * <p>
+     * Override this if there is a more efficient way than first collecting the snapshot and then discarding it.
+     */
+    default MetricSnapshots collect(Predicate<String> includedNames, PrometheusScrapeRequest scrapeRequest) {
+    	MetricSnapshots allSnapshots = collect(scrapeRequest);
+        MetricSnapshots.Builder result = MetricSnapshots.builder();
+        for (MetricSnapshot snapshot : allSnapshots) {
+            if (includedNames.test(snapshot.getMetadata().getPrometheusName())) {
+                result.metricSnapshot(snapshot);
+            }
+        }
+        return result.build();
```

Comment:
Only the first line differs from the `collect()` method without the `scrapeRequest` parameter. What about creating a helper method for the common code?